src/gax-internal/src/observability/http_tracing.rs

```diff
@@ -410,32 +433,87 @@ mod tests {
         assert_eq!(*attributes, expected_attributes);
     }
 
+    #[test_case(FromRequestTestCase {
+        method: Method::GET, url: "https://example.com/test", template: None, instrumentation: None, prior_attempt_count: 0,
+        expected_otel_name: "GET", expected_url_full: "https://example.com/test", expected_server_address: "example.com", expected_server_port: 443,
+        expected_url_scheme: Some("https"), expected_url_template: None, expected_url_domain: None }; "basic get")]
+    #[test_case(FromRequestTestCase {
+        method: Method::POST, url: "http://localhost:8080/", template: None, instrumentation: None, prior_attempt_count: 0,
+        expected_otel_name: "POST", expected_url_full: "http://localhost:8080/", expected_server_address: "localhost", expected_server_port: 8080,
+        expected_url_scheme: Some("http"), expected_url_template: None, expected_url_domain: None }; "basic post")]
+    #[test_case(FromRequestTestCase {
+        method: Method::PUT, url: "https://example.com/items/123", template: Some("/items/{item_id}"), instrumentation: None, prior_attempt_count: 0,
+        expected_otel_name: "PUT /items/{item_id}", expected_url_full: "https://example.com/items/123", expected_server_address: "example.com", expected_server_port: 443,
+        expected_url_scheme: Some("https"), expected_url_template: Some("/items/{item_id}"), expected_url_domain: None }; "put with template")]
```

Comment:
I recommend having a test like `fn test_http_span_info_from_request_with_template` where you only verify fields related to the template.

And same for prior attempt.

That might help you flatten out the cases and have tests that are easier to grok.